streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/AssignmentInfo.java

```diff
@@ -130,8 +157,7 @@ public int latestSupportedVersion() {
      */
     public ByteBuffer encode() {
         final ByteArrayOutputStream baos = new ByteArrayOutputStream();
-
-        try (final DataOutputStream out = new DataOutputStream(baos)) {
+        try (final DataOutputStream out = new DataOutputStream(new GZIPOutputStream(baos))) {
```

Comment:
This will break backward compatibility as encoding of earlier versions (<=3) should not change. We can leave version-bytes uncompressed and can compress bytes following version-bytes for v4. This way we have flexibility to choose completely different compression/encoding for later versions (v5, v6..)